mock/private/check.rkt

```diff
@@ -11,8 +11,12 @@ provide
     check-mock-num-calls (-> exact-nonnegative-integer? mock? void?)
 
 
-(define-simple-check (check-mock-called-with? args mock)
-  (mock-called-with? args mock))
+(define-check (check-mock-called-with? args mock)
+  (let ([result (mock-called-with? args mock)])
+    (unless result
+      (fail-check
+        (format "Expected call: ~v\nActual calls: ~v" args (mock-calls mock))))))
```

Comment:
Rather than putting this information in the message, I think it would be better to use `with-check-info` to add the expected and actual and just have the message read "Actual arguments did not match expected" or the like. This cooperates with the formatting that rackunit uses for error messages.